streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -821,6 +821,33 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester
     checkpointWriter.stop()
   }
 
+  test("SPARK-6847: stack overflow when updateStateByKey is followed by a checkpointed dstream") {
+    ssc = new StreamingContext(master, framework, batchDuration)
```

Comment:
Would be nice to add a bit of context in the state, with may be a ASCII art showing the DAG structure